gnocchi/rest/influxdb.py

```diff
@@ -0,0 +1,267 @@
+
+line_protocol = (
+    measurement +
+    # Tags
+    pyparsing.Optional(pyparsing.Suppress(",") +
+                       pyparsing.delimitedList(
+                           pyparsing.OneOrMore(
+                               pyparsing.Group(
+                                   tag_key +
+                                   pyparsing.Suppress("=") +
+                                   tag_value), ",")).setParseAction(
+                                       lambda s, loc, tok: dict(list(tok))),
+                       default={}) +
+    pyparsing.Suppress(" ") +
+    # Fields
+    pyparsing.delimitedList(
+        pyparsing.OneOrMore(
+            pyparsing.Group(field_key +
+                            pyparsing.Suppress("=") +
+                            field_value), ",")).setParseAction(
+                                lambda s, loc, tok: dict(list(tok))) +
+    # Timestamp
+    pyparsing.Optional(pyparsing.Suppress(" ") + timestamp, default=None)
+).leaveWhitespace()
+
+
+query_parser = (
+    pyparsing.Suppress(pyparsing.CaselessLiteral("create")) +
+    pyparsing.Suppress(pyparsing.CaselessLiteral("database")) +
+    pyparsing.Suppress(pyparsing.White()) +
+    (pyparsing.QuotedString('"', escChar="\\") |
+     pyparsing.Word(pyparsing.alphas + "_",
+                    pyparsing.alphanums + "_")) +
+    pyparsing.Suppress(
+        pyparsing.Optional(pyparsing.Optional(pyparsing.White()) +
+                           pyparsing.Optional(pyparsing.Literal(";"))))
+)
+
+
+class InfluxDBController(rest.RestController):
+    _custom_actions = {
+        'ping': ['HEAD', 'GET'],
+        'query': ['POST'],
+        'write': ['POST'],
+    }
+
+    DEFAULT_TAG_RESOURCE_ID = "host"
+
+    @pecan.expose()
+    def ping(self):
+        pecan.response.headers['X-Influxdb-Version'] = (
+            "Gnocchi " + pbr.version.VersionInfo('gnocchi').version_string()
+        )
+
+    @pecan.expose('json')
+    def post_query(self, q=None):
+        if q is not None:
+            try:
+                query = query_parser.parseString(q)
+            except pyparsing.ParseException:
+                api.abort(501, {"cause": "Not implemented error",
+                                "detail": "q",
+                                "reason": "Querying is not implemented"})
+            resource_type = query[0]
+            api.enforce("create resource type", {"name": resource_type})
+            schema = pecan.request.indexer.get_resource_type_schema()
+            rt = schema.resource_type_from_dict(resource_type, {}, 'creating')
+            try:
+                pecan.request.indexer.create_resource_type(rt)
+            except indexer.ResourceTypeAlreadyExists:
+                pass
+            pecan.response.status = 204
+
+    @staticmethod
+    def _write_get_lines():
+        encoding = pecan.request.headers.get('Transfer-Encoding', "").lower()
+        if encoding == "chunked":
+            if uwsgi is None:
```

Comment:
how usable is this without chunked? if it isn't, i don't think we should support this especially since we're checking this constantly.

Reply:
it's completely usable without chunked encoding. That's what the tests do.

But real InfluxDB client sends chunked encoding.

Reply:
so unchunked is only for tests?

Reply:
In real life, really likely, until someone starts to talk InfluxDB to Gnocchi in regular encoding (but I don't think InfluxDB does that?)